core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1098,7 +1098,14 @@ private[spark] class DAGScheduler(
   private def submitMissingTasks(stage: Stage, jobId: Int) {
     logDebug("submitMissingTasks(" + stage + ")")
 
-    // First figure out the indexes of partition ids to compute.
+    // Before find missing partition, do the intermediate state clean work first.
+    stage match {
+      case sms: ShuffleMapStage if stage.isIndeterminate =>
```

Comment:
shall we add `if stage.latestInfo.attemptNumber() > 0` as well?

Reply:
It will cause the correctness problem because we didn't call `makeNewStageAttempt` so far here.